src/extension/background-script/actions/setup/setIcon.ts

```diff
@@ -0,0 +1,59 @@
+import browser, { Runtime } from "webextension-polyfill";
+import { SetIconMessage } from "~/types";
+
+import state from "../../state";
+
+const tabIcons = new Map<number, string>();
+
+enum ExtensionIcon {
+  Default = "alby_icon_yellow",
+  Tipping = "alby_icon_blue",
+  Active = "alby_icon_green",
+}
+
+const setIconMessageHandler = async (
+  message: SetIconMessage,
+  sender: Runtime.MessageSender
+): Promise<{ data: boolean }> => {
+  // Under some circumstances a Tab may not be assigned an ID
+  if (!sender.tab || !sender.tab.id) {
+    return Promise.resolve({
+      data: false,
+    });
+  }
+
+  await setIcon(message.args.icon, sender.tab.id);
+
+  return Promise.resolve({
+    data: true,
+  });
+};
+
+const setIcon = async (icon: string, tabId: number): Promise<void> => {
+  const currentIcon = tabIcons.get(tabId);
+
+  // The active icon has priority over tipping
+  if (
+    currentIcon &&
+    currentIcon == ExtensionIcon.Active &&
```

Comment:
`===` should be prefered because it's type-safe. Should check eslint rules I guess.  
Context https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons